rotkehlchen/globaldb/updates.py

```diff
@@ -643,14 +653,21 @@ def perform_update(
                 log.info('Finishing assets update. Replacing users globaldb with the updated information')  # noqa: E501
                 _replace_assets_from_db(GlobalDBHandler().conn, tmpdir / temp_db_name)
 
-            return None
+        with GlobalDBHandler().conn.critical_section():
+            # We apply in a different step since the spam assets don't require conflict resolution
+            # and make use of the logic in the globaldb class so the final db has to be used.
+            # Use a critical section to avoid another greenlet adding the same asset but is fine
+            # since we don't make any external query.
```

Comment:
This comment has some wording mistakes but also does not make sense. If you don't need a critical section, don't do it. If you need do it. Being future proof is also a need.

```suggestion
            # Applied in a different step since the spam assets don't require conflict resolution
            # and utilize logic of the globaldb class so the final db has to be used.
            # Use a critical section to avoid another greenlet adding spam assets at the same time
```